src/Meditation.UI/Services/AttachedProcessContext.cs

```diff
@@ -0,0 +1,24 @@
+﻿using Meditation.AttachProcessService;
+using System;
+
+namespace Meditation.UI.Services
+{
+    internal class AttachedProcessContext : IAttachedProcessProvider, IAttachedProcessController
+    {
+        public event Action<IProcessSnapshot?>? AttachedProcessChanged;
+
+        public IProcessSnapshot? ProcessSnapshot { get; private set; }
+
+        public void Attach(IProcessSnapshot processSnapshot)
+        {
+            ProcessSnapshot = processSnapshot;
+            AttachedProcessChanged?.Invoke(processSnapshot);
+        }
+
+        public void Detach()
```

Comment:
[ToFix] I guess we need a static method Attach that can return IDisposable object that will call detach on dispose. It also prevent call detach before attach or call attach twice.

Reply:
I refactored it and added checks and dialogs that notify user when they are doing something invalid.

This is a shared dependency of a couple classes. Changing the `Attach` to static would allow to attach multiple processes at once, which is currently not supported and I don't see a use case for it.